circuit_knitting/cutting/cut_finding/cco_utils.py

```diff
@@ -138,23 +138,25 @@ def greedy_best_first_search(
         Callable, search_space_funcs.next_state_func
     )
 
-    if search_space_funcs.goal_state_func(state, *args):
-        return state
-
-    best = min(
-        [
-            (search_space_funcs.cost_func(next_state, *args), k, next_state)
-            for k, next_state in enumerate(
-                search_space_funcs.next_state_func(state, *args)
-            )
-        ],
-        default=(None, None, None),
-    )
-
-    if best[-1] is None:  # pragma: no cover
-        # This covers a rare edge case.
-        # We have so far found no circuit that triggers it.
-        # Excluding from test coverage for now.
-        return None
-
-    return greedy_best_first_search(best[-1], search_space_funcs, *args)
+    while True:
+        if search_space_funcs.goal_state_func(state, *args):
```

Comment:
It seems we don't expect this to be `None`. That is the rare case below. So in the case where we aren't ready to return yet, we expect this to be `[]` or what here? Maybe we can explicitly state it in the `if` statement.

Reply:
OK sorry I got a little mixed up. Disregard above.